tab_mur_data_prep.Rmd

```diff
@@ -0,0 +1,167 @@
+to work with, I am only using P4, and have excluded the Liver and spleen samples. 
+
+```{r}
+# Load ggplot2 library
+library(ggplot2)
+
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+Set up the data
+
+```{r}
+# Read in data
+all.data <- readRDS(file.path("tab_mur_data", "counts_tab_mur.RDS"))
+
+# Keep the gene names separate
+gene.names <- all.data$genes
+
+# Get rid of gene column
+all.data <- all.data[, -1]
+```
+ 
+Make cell info dataframe for labeling purposes
+
+```{r}
+# Clean up sample names
+colnames(all.data) <- gsub("_possorted_genome.bam", "", colnames(all.data))
+
+# Make a dataframe of this cell info
+cell.info <- data.frame(strsplit(colnames(all.data), "\\.")) %>% 
+  t() 
+
+# Change column names
+colnames(cell.info) <- c("cell_barcode", "sample.id")
+
+# Keep the full id names for each as well for matching purposes
+cell.info <- data.frame(full.id = colnames(all.data), cell.info)
+```
+
+Read in the metadata file obtained from figshare, clean it up so we can merge it
+with the sample names obtained in the cell.info dataframe. 
+
+```{r}
+# Import metadata file and make it with only the variables and samples we want
+metadata <- readr::read_csv(file.path("tab_mur_data", "metadata_droplet.csv")) %>% 
+  dplyr::select("channel", "mouse.id", "tissue") %>% 
+  dplyr::filter(grepl("10X_P4", channel), tissue != "Liver", tissue != "Spleen") 
+
+# Make column name matching so we can use inner_join to merge the info
+colnames(metadata)[1] <- "sample.id"
+```
+
+The liver samples had 95% of their data dropped in Alevin due to noisy cell 
+barcodes. 
+Spleen is excluded because then it became the only tissue that wasn't represented
+in both mice. 
+
+```{r}
+# Merge these two frames together
+metadata <- dplyr::inner_join(cell.info, metadata, by = "sample.id") %>%
+  readr::write_tsv("metadata.tsv")
+```
+
+Make a data matrix for ease of collecting stats
+
+```{r}
+data.mat <- all.data %>% as.matrix()
+```
+
+For least stringent filter, let's determine what the smallest expression value is
+
+```{r}
+# Determine a detection cutoff
+min.cutoff <- min(data.mat[which(data.mat != 0)])
+```
+
+Let's make a detection matrix to just hold each data point as expressed or not
+
+```{r}
+# Make a matrix of detected or not detected logical values
+boolean <- data.mat >= min.cutoff
+```
+
+Determine cutoff for how many samples should express a gene 
+
+```{r}  
+# Make a vector that has the number of samples that express a gene
+gene.sum <- apply(boolean, 1, sum)
+
+# Plot this information
+plot(density(gene.sum), xlim = c(0, 300));
+abline(v = 3)
+```
+
+Determine cutoff for how many genes should be expressed in a sample 
+
+```{r}
+# Make vector that has the number of genes expressed by each cell
+sample.sum <- apply(boolean, 2, sum) 
+
+# Put this with the cell.info so we can look at the distributions by samples
+sample.sum.df <- data.frame(sample.sum, cell.info) 
+
+# Plot with each sample as its own line
```

Comment:
Would it be useful to include _tissue_ information in this plot? Either via facets or line types or something like that?